src/components/molecules/search-bar.tsx

```diff
@@ -1,76 +1,50 @@
-import React, { useCallback, useContext, useRef, useState } from "react"
+import { globalHistory } from "@reach/router"
+import React, { useState } from "react"
 import { useHotkeys } from "react-hotkeys-hook"
-import { InterfaceContext } from "../../context/interface"
 import OSShortcut from "../atoms/os-shortcut"
-import TextInput from "../atoms/text-input"
 import SearchIcon from "../fundamentals/icons/search-icon"
+import SearchModal from "../templates/search-modal"
 
 const SearchBar: React.FC = () => {
-  const { onSearch, display } = useContext(InterfaceContext)
-  const [query, setQuery] = useState("")
-  const [focusing, setFocusing] = useState(false)
-  const searchRef = useRef(null)
+  const [showSearchModal, setShowSearchModal] = useState(false)
 
-  const hotKeyFocus = () => {
-    if (searchRef && searchRef.current) {
-      setFocusing(true)
-      searchRef.current.focus()
-      return false
-    }
+  const toggleSearch = (e) => {
+    e.preventDefault()
+    e.stopPropagation()
+    setShowSearchModal((show) => !show)
   }
 
-  const onKeyDown = useCallback(
-    event => {
-      switch (event.key) {
-        case "Enter":
-          event.preventDefault()
-          event.stopPropagation()
-          if (onSearch) {
-            onSearch(query)
-          }
-          searchRef.current.blur()
-          break
-        case "Esc":
-        case "Escape":
-          searchRef.current.blur()
-          break
-        default:
-          break
-      }
-    },
-    [onSearch, query]
-  )
-
-  const handleChange = e => {
-    if (focusing) {
-      setFocusing(false)
-      return
-    } else {
-      setQuery(e.target.value)
-    }
+  const closeModal = () => {
+    setShowSearchModal(false)
   }
 
-  useHotkeys("cmd+k", hotKeyFocus, {}, [searchRef])
-  useHotkeys("ctrl+k", hotKeyFocus, {}, [searchRef])
+  useHotkeys("cmd+k", toggleSearch, {}, [])
+  useHotkeys("ctrl+k", toggleSearch, {}, [])
+
+  React.useEffect(() => {
+    return globalHistory.listen(({ action }) => {
+      if (action === "PUSH") {
+        closeModal()
+      }
+    })
+  }, [])
 
   return (
-    <div className="flex items-center">
-      {display ? (
-        <>
-          <SearchIcon className="text-grey-40" />
-          <div className="mr-xsmall ml-base">
-            <OSShortcut macModifiers="⌘" winModifiers="Ctrl" keys="K" />
-          </div>
-          <TextInput
-            value={query}
-            onKeyDown={onKeyDown}
-            onChange={handleChange}
-            ref={searchRef}
-            placeholder="Search anything..."
-          />
-        </>
-      ) : null}
-    </div>
+    <>
+      <button
```

Comment:
Should this be wider you think? 

It feels a bit off when you can only click the small text and symbol in the upper left, we could definitely also leave it as is without any issues.

Reply:
totally! good catch :+1: